lib/tasks/staging_seeds.rake

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+namespace :comfy do
+  
+  # Locally stored seeds - assumes you already have the local folder - 
+  # it won't create it 
+  LOCAL = (ComfortableMexicanSofa.config.seeds_path + '/protected-planet').freeze
+  REMOTE = 'ProtectedPlanet/current/db/cms_seeds/protected-planet'.freeze
+  PP_STAGING = 'new-web.pp-staging.linode.protectedplanet.net'.freeze
+  PP_USER = 'wcmc'.freeze
+
+  desc "Import CMS Seed data from staging. Can be run with argument or can accept user input if no argument is supplied"
+  task :staging_import, [:seed_type] => [:environment] do |_t, args| 
+    answer = args[:seed_type]
+    
+    if answer.nil?
+      puts question = "What would you like to import? 'All/Files/Layouts/Pages' or 'Nothing' to quit"
+      valid_answers = ['All', 'Files', 'Layouts', 'Pages', 'Nothing']
+      answer = STDIN.gets.chomp.downcase.capitalize
+      until valid_answers.include?(answer)
+        puts question  
+      end
+      
+      # Fast-tracked unhappy path
+      abort('Goodbye') if answer == 'Nothing'
+    else
+      answer = answer.downcase.capitalize!
+    end
+    
+    puts "Importing CMS Seed data from Staging Folder to #{LOCAL} ..."
+
+    new_session =  SyncSeeds.new(PP_STAGING, PP_USER)
+
+    # SSH into staging server with Net::SSH
+    new_session.start_session do |session|
+      # First get rid of any local top-level (i.e. which exist in the main 
+      # directory of REMOTE) folders/files that don't exist remotely
+      new_session.compare_folders('*', LOCAL, REMOTE)
+
+      local_list = new_session.list_local_files(LOCAL)
+      remote_list = new_session.list_remote_files(REMOTE)
+
+      if answer == 'All'
+        new_session.main_task(local_list, remote_list)
+      else
+        local_list.filter! { |f| f == answer.downcase } 
+        remote_list.filter! do |f|
+          f.name.force_encoding('UTF-8') == answer.downcase
+        end
+
+        puts "Downloading a new set of #{answer.downcase}..."
+
+        new_session.main_task(local_list, remote_list)
+      end
+
+      puts "Finished downloads, now replacing your local seed data with your selection..."
+
+      new_session.commence_import(answer) 
```

Comment:
I was under the impression this would have to go into a different task altogether with staging_import one?
The idea was to leave this task to just fetch the data from the server and then create another rake task that would do both things. The alternative would be to pass another argument to this task but I feel it might be too much.

Also I think it might be a good idea to first check that new files have been actually fetched from the server before running the importer, so it doesn't run unnecessarily if everything is already up to date. Even though it seems the importer is smart enough to notice changes by itself while running. What do you think?

Reply:
Re the different task, when I first conceived how I would go about this Rake task I was thinking to just download them like you said, and then give the user the option to run the import if they wanted to. I guess it came down to my assumption that if you're intending to run this task you're doing so with the intention of wanting to update your database as well, so I thought I'd just save time by importing automatically.

Re the second point, that is a good shout. However, in an earlier comment, you addressed the point of whether I should download individual files if they're missing or older, or the entire folder to reduce the number of `scp` commands being run. Is that still valid? To make the importer smarter it could just download files instead of the whole folder again in the case of just one file or folder (in `files`, `pages` etc - just has to be in one of the directories in `cms_seeds`) - at the moment I've put in a `break` to stop it from downloading the whole folder again if `check` returns `true`, but I feel that it is a more natural way to just check each file individually.

Reply:
Yeah, I guess we could leave it as it is then. Not particuarly relevant as you say.

I'm not sure I follow abou the second point. What I had in mind was about adding a flag to store the result of the whole fetching process from the staging server, if possible. If there have been changes then launch the import as well, don't otherwise. But again, this is also a minor thing considering that the importer seems to deal with this already in a way